vnext/ReactUWP/Views/FlyoutViewManager.cpp

```diff
@@ -155,8 +156,43 @@ void FlyoutShadowNode::createView() {
   m_flyoutClosedRevoker =
       m_flyout.Closed(winrt::auto_revoke, [=](auto &&, auto &&) {
         auto instance = wkinstance.lock();
-        if (!m_updating && instance != nullptr)
+        if (!m_updating && instance != nullptr) {
+          // When the flyout closes, attempt to move focus to
+          // its anchor element to prevent cases where focus can land on
+          // an outer flyout content and therefore trigger a unexpected flyout
+          // dismissal
+          winrt::FocusManager::TryFocusAsync(
+              m_targetElement, winrt::FocusState::Programmatic);
+
           OnFlyoutClosed(*instance, m_tag, false);
+        }
+      });
+
+  // Turn off AllowFocusOnInteraction so that focus cannot land in the top
+  // content element of the flyout. If focus lands in the flyout presenter, then
+  // on moving focus somewhere else, including the children on the flyout
+  // presenter, the flyout can dismiss.
+  //
+  // However, turning AllowFocusOnInteraction off at the root of the flyout and
+  // flyout presenter turns it off for all children of the flyout. In order to
+  // make sure that interactions with the content of the flyout still work as
+  // expected, AllowFocusOnInteraction is turned on for all children of the
+  // flyout presenter.
+  m_flyout.AllowFocusOnInteraction(false);
+
+  m_token = m_flyout.RegisterPropertyChangedCallback(
+      winrt::Flyout::ContentProperty(),
+      [=](winrt::DependencyObject sender, winrt::DependencyProperty dp) {
+        if (auto flyout = sender.try_as<winrt::Flyout>()) {
+          if (auto content = flyout.Content()) {
+            auto children = content.GetChildrenInTabFocusOrder();
+
+            for (auto child : children) {
+              if (auto fe = child.try_as<winrt::FrameworkElement>())
+                fe.AllowFocusOnInteraction(true);
```

Comment:
>AllowFocusOnInteraction(true) [](start = 19, length = 29)

I'm concerned that, at some point, we may be forcing this property to true when some other codepath was explicitly setting this false.  Is there a way to determine whether or not it was set by default, and only flip it true if so?  ..or should we expose this property on the ReactNative side and allow users to take control of the problem?

Reply:
I don't think there's a way to know if the value of AllowFocusOnInteraction is set to its default value or is an override. Even if there was, it's possible that in this case, it would consider the value of AllowFocusOnInteraction to be an "override," as the default value of the prop on these elements is being overridden by the value set on its parent. FWIW, since before this change the flyout and the flyout presenter had its allowFocusOnInteraction property set to true, all its children did as well (verified in master). That's not to say your worry wouldn't happen, but this change is following the "default" behavior, minus the flyout and flyout presenter having their allowFocusOnInteraction set to false.

I'm not sure if it makes sense to get the platform to expose this property, though it might be something we need later when we get to more commanding-heavy UI. I can attempt a different solution of just trying to avoid focus landing on the flyout in general and seeing where that goes, and if that's successful, that would avoid this kind of change.